mainapp/templates/mainapp/contributor_form.html

```diff
@@ -23,7 +23,13 @@
   </form>
 <script>
     window.onload = function() {
-        $("#id_phone").attr('type', 'number');
+        $("#id_phone").attr({'type':'tel',
+                'pattern':'[0-9]{10}'}).keypress(function (e) {
+     //if the letter is not digit then display error and don't type anything
+     if (e.which != 8 && e.which != 0 && (e.which < 48 || e.which > 57)) {
+               return false;
```

Comment:
It would help users if we respond with some response instead of nothing.

Reply:
Character key can be pressed only in desktops or laptops. On mobile devices it should switch to Numeric keypad automatically. I dont think a prompt is necessary.

Reply:
No response while typing could leave the user on confusion whether his machine or connection has stalled. If we're ignoring user input, It's always better to acknowledge it to the user.

Reply:
hmmm I understand. Then what do you propose . I dont think an alert will be a good idea. And anything else can hamper the ui.

Reply:
Any way there should be backend verification as well. We can never rely on front end validation

Reply:
I have a fix for this in #484